### DISCUSSION TITLE
the map in the props of the AdvancedMarker component.

### DISCUSSION THREAD

**Discussion by abb438** [visgl/react-google-maps#280] [Q&A]
I'm using translation software to create the text, so sorry if there are any oddities.
AdvancedMarkerComponent receives AdvancedMarkerElementOptions type as props and can pass a map, but the code does not reference the map in the props, but instead sets up a const map in the component.
If I want to display it at an arbitrary time without displaying it at first, I can pass null as a value, but it will be displayed once.
I think I need to set markerRef.map = null in useEffect etc. after that, but am I using it wrong?
myCode
```
<AdvancedMarker
      map={null}
      position={position}
      ref={setMarkerRef}
    >
      {children}
 </AdvancedMarker>
```
Library Code
```
function useAdvancedMarker(props: AdvancedMarkerProps) {
  const [marker, setMarker] =
    useState<google.maps.marker.AdvancedMarkerElement | null>(null);
  const [contentContainer, setContentContainer] =
    useState<HTMLDivElement | null>(null);

  const map = useContext(GoogleMapsContext)?.map;
  const markerLibrary = useMapsLibrary('marker');

  const {
    children,
    className,
    onClick,
    onDrag,
    onDragStart,
    onDragEnd,
    collisionBehavior,
    draggable,
    position,
    title,
    zIndex
  } = props;
  
  useEffect(() => {
    if (!map || !markerLibrary) return;

    const newMarker = new markerLibrary.AdvancedMarkerElement();
    newMarker.map = map;
```

**Comment by usefulthink** [comment]
Actually, the map-prop needs to be removed from the AdvancedMarkerProps type, since assigning the map object to the marker instance is handled automatically.

We generally assume that all markers that are rendered should also be displayed on the map, If you want to show a marker at a later point, just don't render it when it shouldn't be shown, something like this:

```jsx
const MyMap = () => {
  return (<Map {...mapProps}>
    {shouldShowMarker && <AdvancedMarker {...advancedMarkerProps}>Marker Content</AdvancedMarker>}
  </Map>);
}
```

**Reply by abb438** [reply]
Thanks for the reply.
map is not good enough to remain in advancedMarkerProps and that the actual behavior is correct as it is now.

**Reply by usefulthink** [reply]
Not so much "is not good enough", more like "using it leads to unexpected errors"

**Comment by usefulthink** [comment]
added a bug to adress this: #284
